src/cli/ask.rs

```diff
@@ -0,0 +1,124 @@
+use crate::cli::llm_client::LLMClients;
+use crate::cli::util::read_openai_api_key;
+use crate::state::State;
+use crate::OpenAIClient;
+use std::sync::{Arc, Mutex};
+use tokio::runtime::Runtime;
+use tokio::select;
+
+use crate::CtrlcFuture;
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, IntoPipelineData, PipelineData, ShellError, Signature, SyntaxShape, Value,
+};
+
+#[derive(Clone)]
+pub struct Ask {
+    state: Arc<Mutex<State>>,
+}
+
+impl Ask {
+    pub fn new(state: Arc<Mutex<State>>) -> Self {
+        Self { state }
+    }
+}
+
+impl Command for Ask {
+    fn name(&self) -> &str {
+        "ask"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("ask")
```

Comment:
I wonder if this is the right command name for this? Not that I have anything better...

Reply:
I'm a fan of ask, seems fairly clear what it does and has the advantage of being short